src/main/java/com/hazelcast/kubernetes/DefaultKubernetesClient.java

```diff
@@ -230,6 +241,14 @@ private static Integer getPortFromEndpointAddress(JsonValue endpointAddressJson)
         return result;
     }
 
+    private static String parseNodeName(JsonObject json) {
+        return toString(json.get("spec").asObject().get("nodeName"));
+    }
+
+    private static String parseZone(JsonObject json) {
+        return toString(json.get("metadata").asObject().get("labels").asObject().get("failure-domain.beta.kubernetes.io/zone"));
```

Comment:
label name is beta. Does that mean that we have to change it in future when it is out of beta?

Reply:
Good question! In kubernetes > 1.13, it's already out from beta, so it's named: "failure-domain.kubernetes.io/zone". But 1.13 is not available yet anywhere (GKE, Azure, etc.). One solutions is that we read upfront both labels (giving priority to non-beta: "failure-domain.kubernetes.io/zone"). Wdyt?

Reply:
I think that is a good solution and would save us from compatibility hell in future.

Reply:
Added.